pkg/registry/generic/registry/store.go

```diff
@@ -387,6 +440,39 @@ var (
 	errDeleteNow       = fmt.Errorf("delete now")
 )
 
+// return if we need to update the finalizers of the object
+func shouldUpdateFinalizers(accessor meta.Object, options *api.DeleteOptions) bool {
```

Comment:
No, it's "update", not "new", because we may need to remove the orphan finalizer if options suggests so.

Also it seems useless to return the diff, because we need to go through the finalizers in the GuaranteedUpdate() anyway, because the finalizers could be changed by another thread.